CDAP-18010: Include namespace in ArtifactDescriptor

wyzhang opened the pull request:
This is part 1 of isolating artifact inspection.

Why:
Running artifact inspection in isolation requires artifact ids
for fetching them from appfabric. ArtifactDescriptor is available
in the artifact inspection process but it doesn't contain namespace.

tivv commented on cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/ArtifactDescriptor.java:
```diff
@@ -26,6 +26,7 @@
  * then by name, and finally by version.
  */
 public final class ArtifactDescriptor implements Comparable<ArtifactDescriptor> {
+  private String namespace;
```
`final`

wyzhang replied:
Fixed.

tivv commented on cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/ArtifactDescriptor.java:
```diff
@@ -26,6 +26,7 @@
  * then by name, and finally by version.
  */
 public final class ArtifactDescriptor implements Comparable<ArtifactDescriptor> {
```
Also please update toString/hashCode/equals/compareTo

wyzhang replied:
Fixed.

wyzhang committed:
CDAP-18010: Include namespace in ArtifactDescriptor

This is part 1 of isolating artifact inspection.

Why:
Running artifact inspection in isolation requires artifact ids
for fetching them from appfabric. ArtifactDescriptor is available
in the artifact inspection process but it doesn't contain namespace.
